### DISCUSSION TITLE
Mapping capslock to ESC on tap and LCtrl on hold

### DISCUSSION THREAD

**Discussion by minhduc0711** [DreymaR/BigBagKbdTrixPKL#64] [Q&A] [answered]
I'm trying to configure what is in title, and I have this in `EPKL_Layouts_Override.ini`:
```
[layout]
QWCLK = ESCAPE/LCONTROL       VKey
```
But after refreshing EPKL, the capslock key is still mapped to `Extend`. Could someone tell me what I did wrong?

**Comment by DreymaR** [comment, marked as answer]
– Did you remember to put your mapping above the existing one, or comment out the existing one? EPKL will normally use the first (active) setting under each section.

[edit: Oh hang on, in your example the mapping is indeed right under the `[layout]` section tag. All well then. 
I'll leave the above reply in for the benefit of future readers.]

– I'm sorry if this is badly explained in the docs, but to map to a dual-function Tap-or-Mod key you have to make the second part its AHK modifier name. In this case, that isn't the VK code `LCONTROL` nor EPKL's name for it such as `vcLCT`, but AHK's name `LCtrl`. I tried it out, and it works.

I guess I ought to allow more aliases for the modifiers in the near future, to prevent such confusion!

**Reply by minhduc0711** [reply]
Thanks, but it seems like I'm having a different problem: my setting is not being picked up at all even if I change the `Layouts_Default` config file. Here's the relevant section:
```
[layout]
;shiftStates    =			0	:	1	:	2	:	6	:	7		;	8	:	9	:	e	:	f
;; SC = VKEY    	CStat	Base	Shift	Ctrl	AltGr	AGrSh	;	SwiSh	SwShS	SwAGr	SwAGS

;;  *** Several options are shown for the Caps key. Of course, these could also be used on other keys. The MoDK is awesome!
;QWCLK = qwBSP    	SKey 											; SC03a: CapsLock as Backspace (by VK: CAPITAL -> BACK)
;QWCLK = Extend   	Mod 											; SC03a: CapsLock as the Extend modifier
;QWCLK = BACK/Ext 	VKey 											; SC03a: CapsLock as a Tap-or-Mod key: Backspace on tap, Extend modifier on hold
;QWCLK = BACK/Ext 	0   	@ex0	@ex1	*#. 	@ex6	@ex7	; SC03a: Caps as Mother-of-DKs (MoDK) on tap, Extend on hold; Ctrl+{Ext} = emoji picker
QWCLK = ESCAPE/LCtrl VKey 		; Generated by the EPKL KeyMapper UI, 2023-07-07_09-47
```
I even tried `QWCLK = qw_A SKey` just to see if the setting is picked up, but the logs apparently show that CapsLock is still mapped to Extend:
```
14  03A	h	d	1.30	CapsLock       	
14  03A	h	u	0.08	CapsLock       	
08  00E	i	d	0.00	Backspace      	
74  03F	h	d	0.80	F5             	
74  03F	i	d	0.00	F5   
```

**EDIT**: a reinstallation fixed everything, perhaps I have messed up the configs somewhere. However, I was still unable to get the tap key (`ESCAPE`) to work, while the hold key (`LCtrl`) does work. Am I using the wrong name for Escape?

**Comment by DreymaR** [comment]
> **EDIT**: a reinstallation fixed everything, perhaps I have messed up the configs somewhere. However, I was still unable to get the tap key (`ESCAPE`) to work, while the hold key (`LCtrl`) does work. Am I using the wrong name for Escape?

Glad it (mostly) worked for you! But your syntax is right. I just tried out your `QWCLK = ESCAPE/LCtrl VKey` line in my own Layouts_Override file (actually copy-pasted it to be sure!) and it works perfectly for me.

No idea why it shouldn't for you. Maybe try out some other tacks like mapping another key to `ESCAPE VKey` just to test that, then to `ESCAPE/LCtrl`? Btw, the `vcESC` syntax should work too.

On a side note, it's not a bad idea you have there. Seems quite practical. If you have an Extend key elsewhere, that is... I'd never give up the power of Extend for another key or key combo! Heh.
